test/jasmine/tests/bar_test.js

```diff
@@ -407,6 +407,20 @@ describe('Bar.calc', function() {
         assertPointField(cd, 'x', [[1, NaN, NaN, 15]]);
         assertPointField(cd, 'y', [[1, 2, 10, 30]]);
     });
+
+    it('should guard against negative marker.line.width values', function() {
+        var gd = mockBarPlot([{
+            marker: {
+                line: {
+                    width: [2, 1, 0, -1, false, true, null, [], -Infinity, Infinity, NaN, {}]
+                }
+            },
+            y: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+        }], {});
+
+        var cd = gd.calcdata;
+        assertPointField(cd, 'mlw', [[2, 1, 0, 0, 0, 1, 0, 0, 0, Infinity, NaN, NaN]]);
```

Comment:
Does having `NaN` is `calcdata[i][j].mlw` lead to console errors?

Reply:
console errors e.g. when drawing a svg path?

Reply:
I didn't notice any:
[Scatter codepen](https://codepen.io/MojtabaSamimi/pen/bXpGbw?editors=0010)
[Bar codepen](https://codepen.io/MojtabaSamimi/pen/MNyWwQ?editors=0010)

Reply:
But that's a very good question.
Now I am wondering if we should patch this test instead of the `lib.mergeArrayCastPositive` to accept 0 instead of `NaN`?
![shouldCheckForNaN](https://user-images.githubusercontent.com/33888540/61744905-5b354d00-ad66-11e9-9f2c-5ac8033a4148.png)

Reply:
> I didn't notice any:

Ok, but looks like marker pts with `mlw = NaN` render as if they had `mlw = 0`, so I would prefer making `NaN` cast to `0` during calc.

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4063/commits/f9db523c7fd838da60607bd9eabf8e84190b97d9 and fixed `scattegeo` test.